src/v/datalake/translation/partition_translator_v2.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "datalake/translation/deps.h"
+#include "datalake/translation/scheduling.h"
+#include "utils/prefix_logger.h"
+#include "utils/retry_chain_node.h"
+
+namespace datalake::translation {
+
```

Comment:
Why do we need `partition_translator_v2` ? Would it be possible to just change the v1 ?

Reply:
yeah, see TODO in PR comment 🙂

Reply:
done